Lib/test/test_code.py

```diff
@@ -141,6 +147,49 @@ def test_newempty(self):
         self.assertEqual(co.co_name, "funcname")
         self.assertEqual(co.co_firstlineno, 15)
 
+    @cpython_only
+    def test_closure_injection(self):
+        # From https://bugs.python.org/issue32176
+        from types import FunctionType, CodeType
+
+        def create_closure(__class__):
+            return (lambda: __class__).__closure__
+
+        def new_code(c_or_f):
+            '''A new code object with a __class__ cell added to freevars'''
+            c = c_or_f.__code__ if isinstance(c_or_f, FunctionType) else c_or_f
```

Comment:
In this test `new_code()` is called only with the code object.

Reply:
I refactored both this and `add_foreign_method` to only handle the cases needed for the test (the original more general versions came from the reported reproducer for the issue)